lib/pbench/server/indexing_tarballs.py

```diff
@@ -175,13 +191,13 @@ def collect_tb(self):
                 "{} generating list of tar balls to process",
                 error_code["GENERIC_ERROR"].message,
             )
-            # tuple to return the status and return value
+            # Return catch-all error and no tarballs
             return (error_code["GENERIC_ERROR"].value, [])
         else:
             if not tarballs:
                 idxctx.logger.info("No tar balls found that need processing")
 
-        return (error_code["OK"].value, sorted(tarballs))
+        return (error_code["OK"].value, sorted(tarballs, key=lambda x: x.size))
```

Comment:
You should be able to omit the `key` parameter; if not, you should probably get rid of the comment at line 72 which suggests that we have to have the `size` field first....